### ISSUE TITLE
Unsupported field type errors

### ISSUE THREAD

**Issue by codeinabox** [git-zodyac/mongoose#19]
Thank you for creating this, I've been trying it out with the following code:

```ts
import { z } from 'zod'
import { extendZod, zodSchemaRaw } from '@zodyac/zod-mongoose'

extendZod(z)

const schema = zodSchemaRaw(
  z.object({
    name: z.string().optional().default('Bob'),
    age: z.number().default(3).optional(),
    isHappy: z.boolean().optional(),
    birthday: z.date().default(new Date())
  })
)
```

However, it results in the following error:

```
Error: Unsupported field type: class ZodDefault extends ZodType {
    _parse(input) {
        const { ctx } = this._processInputParams(input);
        let data = ctx.data;
        if (ctx.parsedType === ZodParsedType.undefined) {
            data = this._def.defaultValue();
        }
        return this._def.innerType._parse({
            data,
            path: ctx.path,
            parent: ctx,
        });
    }
    removeDefault() {
        return this._def.innerType;
    }
}
 ❯ parseObject node_modules/@zodyac/zod-mongoose/src/index.ts:106:21
 ❯ Proxy.zodSchemaRaw node_modules/@zodyac/zod-mongoose/src/index.ts:95:10
```

**Comment by bebrasmell** [comment]
Oi! Strange, on it

**Comment by bebrasmell** [comment]
Could you please specify the version of the package you're using? I could not reproduce the bug with 2.1.0

<img width="553" alt="image" src="https://github.com/user-attachments/assets/a37c9783-96aa-4106-b1fd-7d9d6bc19349">

**Comment by bebrasmell** [comment]
I've found a bug for CJS, where instanceOf just doesn't do the check properly. It's now fixed in 2.3.1
